# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/linux/hotfix-3c515.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

drivers/net/ethernet/3com/3c515.o: error: objtool: cleanup_module(): Magic init_module() function name is deprecated, use module_init(fn) instead
make[6]: *** [scripts/Makefile.build:203: drivers/net/ethernet/3com/3c515.o] Error 255

--- ./drivers/net/ethernet/3com/3c515.c.vanilla	2025-05-26 19:16:50.055582886 +0200
+++ ./drivers/net/ethernet/3com/3c515.c	2025-05-26 19:27:45.400746652 +0200
@@ -1549,7 +1549,7 @@
 
 
 #ifdef MODULE
-void cleanup_module(void)
+static void corkscrew_cleanup_module(void)
 {
 	while (!list_empty(&root_corkscrew_dev)) {
 		struct net_device *dev;
@@ -1563,4 +1563,5 @@
 		free_netdev(dev);
 	}
 }
+module_exit(corkscrew_cleanup_module);
 #endif				/* MODULE */
